src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -120,15 +144,60 @@ void RaftReplDev::push_data_to_all_followers(repl_req_ptr_t rreq) {
 
     group_msg_service()
         ->data_service_request_unidirectional(nuraft_mesg::role_regex::ALL, PUSH_DATA, rreq->pkts)
```

Comment:
whats the behaviour if the raft log append succeeded but the data push failed for one of the follower ?

Reply:
If data is not pushed to only one follower, leader doesn't get any error. The follower which has not received the data, will do a fetch (after a timeout) and get the data. In the meanwhile quorum would have reached for that log entry and committed (assuming there is one more healthy follower).